self-hosting/README.md

```diff
@@ -0,0 +1,304 @@
+
+### Helpful Docker Commands
+
+#### Get  into a container and view specific service logs 
+
+`docker exec -it medic-os /bin/bash`
+
+##### Once inside container
+
+- view couchdb logs:
+`#less /srv/storage/medic-core/couchdb/logs/startup.log`
+- view medic-api logs:
+`#less /srv/storage/medic-api/logs/medic-api.log`
+- view medic-sentinel logs:
+`#less /srv/storage/medic-sentinel/logs/medic-sentinel.log`
+
+#### View container stderr/stdout log
+
+`# docker logs medic-os`
+`# docker logs haproxy`
+
+#### Other handy commands
+
+- list running containers `#docker ps`
+
+- list all available docker containers with their status `# docker ps -a`
+
+- stop container `# docker stop <container_id>`
+
+- start container `docker start <container_id>`
+
+- list all stoped containers `docker ps -f "status=exited"`
+
+## SSL Certificate Installation
+
+Once your application is up,  you can install your ssl certificates by following the steps below.
+
+- First you need to have your ssl certificate files handy. Both the key file (`ssl.key`) that you sued to generate the certificate and the certificate file (`ssl.crt`) issued by certification authority e.g. letsencrypt. 
+
+### Copy certificates into the medic-os container
+
+```bash
+$sudo docker ps
+$sudo docker cp /path/to/ssl.crt medic-os:/srv/settings/medic-core/nginx/private/ssl.crt
+$docker cp /path/to/ssl.key medic-os:/srv/settings/medic-core/nginx/private/ssl.key
+```
+
+### Edit the nginx configuration file
+
+```bash
+$docker exec -it medic-os /bin/bash
+sed -i "s|default.crt|ssl.crt|" /srv/settings/medic-core/nginx/nginx.conf
+sed -i "s|default.key|ssl.key|" /srv/settings/medic-core/nginx/nginx.conf
+
+```
+
+### Restart services
+
+```bash
+# enter medic-os container:
+$docker exec -it medic-os /bin/bash
+ /boot/svc-restart medic-core nginx
+
+```
+
+### View Nginx Logs
+
+```bash
+# enter medic-os container:
+$docker exec -it medic-os /bin/bash
+cd /srv/storage/medic-core/nginx/logs/ 
+ls
+access.log error-ssl.log error.log startup.og
+```
+
+## Data storage & persistence
+
+Docker containers are [stateless](https://www.redhat.com/en/topics/cloud-native-apps/stateful-vs-stateless) by design.  In order to persist your data when a container restarts you need to specify  the volumes that the container can use to store data. The CHT app stores all it's data in the `/srv` folder.  This is the folder that you need to map to your volume before you spin up your containers. 
+
+Ideally you should map this folder to a volume that is backed up regulaly by your cloud hosting provider.
+
+The exmaple below shows how to map this folder in ubuntu
+
+```bash
+# create the /srv folder 
+$sudo mkdir  /srv 
+
+## mount your volume to this folder 
+$sudo mount /dev/xvdg /srv # the attached volume number varries find your volume by running lsblk
+```
+
+Update your compose file  so that the containers store data to this folder
```

Comment:
If a user goes through all of set up, and then changes the container mount point, it will wipe the data from the install, right? Maybe we should call this out?